### PR TITLE
Include secret name and keys in describe binding 

### LINKED ISSUE #1778: Include info about the injected secret when describing a binding
I'd like to be able to always use svcat and never fall back to kubectl. 😀 So it would be helpful to display information about the associated secret when looking at a binding:

```console
$ svcat describe binding testdb
  Name:        testdb
  Namespace:   default
  Status:      Ready - Injected bind result @ 2018-03-01 16:39:52 +0000 UTC
  Secret:      testdb-creds
  Instance:    testdb

Secret Data:
  database-host:  61 bytes
  database-name:  10 bytes
  password:       16 bytes
  port:           4 bytes
  username:       47 bytes
```

### LINKED ISSUE #1970: Support decoding secret data directly in svcat
#1961 displays the secret name and keys for a binding, but leaves it as an exercise for the reader to figure out how to see the secret data values.

@n3wscott had the awesome suggestion to add a flag to optionally decode and print the secret data in the output.

```console
$ svcat describe binding ups-binding -n test-ns --show-secrets
  Name:        ups-binding
  Namespace:   test-ns
  Status:      Ready - Injected bind result @ 2018-04-19 22:16:00 +0000 UTC
  Secret:      ups-binding
  Instance:    ups-instance

Secret Data:
  special-key-1:  iloveponies
  special-key-2:   andkittiestoo
```

The help text for `--show-secrets` should make it clear that the secret data will be decoded and included in the output.

### PR DESCRIPTION
**Pull request by carolynvs**
- [x] Print secret name in `svcat describe binding`
- [x] Print secret keys when available
- [x] Print error when secret should be available but can't be retrieved
- [x] Happy path output test
- [x] Edge case unit tests, e.g. no error when the binding isn't ready

```console
$ svcat describe binding ups-binding -n test-ns
  Name:        ups-binding
  Namespace:   test-ns
  Status:      ErrorInstanceNotReady - Binding cannot begin because referenced ServiceInstance "ups-instance" is not ready @ 2018-04-19 23:02:37 +0000 UTC
  Secret:      ups-binding
  Instance:    ups-instance


$ svcat describe binding ups-binding -n test-ns
  Name:        ups-binding
  Namespace:   test-ns
  Status:      Ready - Injected bind result @ 2018-04-19 22:16:00 +0000 UTC
  Secret:      ups-binding
  Instance:    ups-instance

Secret Data:
  special-key-1   15 bytes
  special-key-2   15 bytes


$ kubectl delete secret -n test-ns ups-binding
secret "ups-binding" deleted


$ svcat describe binding ups-binding -n test-ns
  Name:        ups-binding
  Namespace:   test-ns
  Status:      Ready - Injected bind result @ 2018-04-19 22:16:00 +0000 UTC
  Secret:      ups-binding
  Instance:    ups-instance

Secret Data:
  unable to get secret test-ns/ups-binding (secrets "ups-binding" not found)
```
Closes #1778
Closes #1970

### CODE REVIEW

**Comment by jeremyrickard** [comment]
This is super cool! LGTM.

File: cmd/svcat/output/binding.go [hunk 1]
```diff
@@ -123,6 +119,28 @@ func WriteAssociatedBindings(w io.Writer, bindings []v1beta1.ServiceBinding) {
...
+	if err != nil {
+		// We should have been able to find a secret but couldn't for some reason,
+		// warn about it without blowing up the entire command.
+		fmt.Fprintf(w, "  %s", err.Error())
+		return
+	}
+
+	t := NewDetailsTable(w)
+	for key, value := range secret.Data {
+		t.Append([]string{key, fmt.Sprintf("%d bytes", len(value))})
```
Comment (n3wscott) [inline cmd/svcat/output/binding.go]: is it a bad idea to show the secret data? Or output the kubectl command to display the data if we are unwilling to?

Comment (carolynvs) [reply cmd/svcat/output/binding.go]: I was just blindly ripping off kubectl but it would be great to have it do the decoding for you. Mind if I add that as a follow-on PR for the new flag?

Comment (carolynvs) [reply cmd/svcat/output/binding.go]: Here's an issue to track adding that flag

https://github.com/kubernetes-incubator/service-catalog/issues/1970

**Comment by carolynvs** [comment]
I had time so I added the `--show-secrets` flag to this PR.

**Comment by jboyd01** [comment]
LGTM, nice addition!
